### ISSUE TITLE
Py3 compatibility

### ISSUE THREAD

**Issue by skorokithakis** [BackMarket/django-admincommand#10]
The currently-published version isn't compatible (and thus installable) with Py3. Could that be added?

**Comment by amirouche** [comment]
I could work on Python 3 port. 

1) What would be the ideal Django versions to support?

2) What about the async support? I don't remember what was the purpose of that feature. As of today, I am not sure how to properly support async (without relying on Celery) or forking another processus.

**Comment by skorokithakis** [comment]
I think the latest versions (anything past 2) would be necessary to support. About async support, I don't think that's as necessary as Py3 support, since the library can be used as-is anyway.

For async support, you could use something like this:

https://django-background-tasks.readthedocs.io/en/latest/

It's simple to set up, it basically only requires one cron job, and works well.

**Comment by amirouche** [comment]
I managed to make django-sneak work with python 3 and django 2. It is definitely possible to port `admincommand` to python3 / django2. That being said, my 2 cents are that this is wrong headed, modern back-offices require more than what django admin has to offer a good example of that is the, so called, async support in django admincommand. I remember now that this was added to django-admincommand to support the execution of long running processus. The thing is that to fully support the execution of django commands in the background, it will require integration with the logging system and capture stdout / stderr somehow and store it in a relevant model. That leads to think that they will be tight coupling between the background job executor and django-admincommand and AFAIK there is no such django application. Better is the enemy of good. You could make the thing less painful if there was some kind of structured logging support, in which case keeping track of the result of the background execution of a command would be as straightforward as `structured_log__app="django_admincommand"` `structured_log__commandname="compute_billing"` kind of queries but again there is no such django application.

tl;dr: good luck :+1:

**Comment by skorokithakis** [comment]
Is this required to get Py3 working?

**Comment by amirouche** [comment]
Le mer. 18 sept. 2019 à 23:36, Stavros Korokithakis <
notifications@github.com> a écrit :

> Is this required to get Py3 working
>
django-admincommand rely on django-sneak to display the django commands in
the admin.

ref: https://github.com/amirouche/django-sneak2

**Comment by skorokithakis** [comment]
And django-sneak isn't yet compatible with 3?

**Comment by amirouche** [comment]
Le mer. 18 sept. 2019 à 23:55, Stavros Korokithakis <
notifications@github.com> a écrit :

> And django-sneak isn't yet compatible with 3?
>

django-sneak2 is compatible with python3 and django2.


-- 
Amirouche ~ amz3 ~ https://hyper.dev

**Comment by skorokithakis** [comment]
Okay so what needs to be done for admincommand to support 3?

**Comment by amirouche** [comment]
I am going to sleep. Let  see  tomorow

**Comment by skorokithakis** [comment]
So can this library support Python 3? Do you accept PRs for that?

**Comment by skorokithakis** [comment]
This is already compatible with 3. I'm not sure why PyPI doesn't list it on 3.
